internal/controllers/rest_controller.go

```diff
@@ -76,19 +129,72 @@ func (rc RESTController) OIDCProviderCallback(w http.ResponseWriter, r *http.Req
 	}
 
 	// If the flow is secure, a JWT will be created...
-	jwt, err := rc.oidcService.CreateJWT(&token.CustomClaims)
+	jwt, err := rc.oidcService.Create(&token.CustomClaims)
 	if err != nil {
 		rc.logger.Error("an error occurred while creating a JWT. Error: " + err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	// ...and then a JSON data to display as body content.
-	jsonBody, err := json.Marshal(jwt)
-	if err != nil {
-		rc.logger.Error("an error occurred while validating some claims. Error: " + err.Error())
-		return
+	// ...set a session cookie.
+	rc.setSessionCookie(w, r, "access_token", jwt.AccessToken)
+
+	rc.logger.Infof("Redirecting back to %s", rc.configuration.AuthServiceURL)
+	http.Redirect(w, r, rc.configuration.AuthServiceURL, http.StatusFound)
+}
+
+// Logout is responsible for revoking a token and deleting an authentication cookie.
+func (rc RESTController) Logout(w http.ResponseWriter, r *http.Request) {
+	// TODO: revoke token
+	rc.deleteCookie(w, "access_token")
+}
+
+// getCredentials is responsible for simply getting credentials.
+func (rc RESTController) getCredentials(r *http.Request) (string, error) {
+	rc.logger.Info("Get request credentials...")
+	for i, auth := range rc.credentials {
```

Comment:
This `for` can be moved to the service, business logic.

Reply:
The controller doesn't need to know the handlers

Reply:
we've improved these conditions, I believe you don't need to move that to service.